satpy/tests/reader_tests/test_viirs_compact.py

```diff
@@ -2435,30 +2437,48 @@ def fill_h5(root, stuff):
             h5f.attrs[attr] = val
         h5f.close()
 
-    def test_get_dataset(self):
-        """Retrieve datasets from a DNB file."""
+        self.client = None
+
+    def _dataset_iterator(self):
         from satpy.readers.viirs_compact import VIIRSCompactFileHandler
         from satpy.tests.utils import make_dataid
 
         filename_info = {}
         filetype_info = {'file_type': 'compact_dnb'}
-        dsid = make_dataid(name='DNB', calibration='radiance')
         test = VIIRSCompactFileHandler(self.filename, filename_info, filetype_info)
-        ds = test.get_dataset(dsid, {})
-        self.assertEqual(ds.shape, (752, 4064))
-        self.assertEqual(ds.dtype, np.float32)
-        self.assertEqual(ds.attrs['rows_per_scan'], 16)
 
+        dsid = make_dataid(name='DNB', calibration='radiance')
+        ds1 = test.get_dataset(dsid, {})
         dsid = make_dataid(name='longitude_dnb')
-        ds = test.get_dataset(dsid, {'standard_name': 'longitude'})
-        self.assertEqual(ds.shape, (752, 4064))
-        self.assertEqual(ds.dtype, np.float32)
-        self.assertEqual(ds.compute().shape, (752, 4064))
-        self.assertEqual(ds.attrs['rows_per_scan'], 16)
+        ds2 = test.get_dataset(dsid, {'standard_name': 'longitude'})
+        dsid = make_dataid(name='latitude_dnb')
+        ds3 = test.get_dataset(dsid, {'standard_name': 'latitude'})
+        dsid = make_dataid(name='solar_zenith_angle')
+        ds4 = test.get_dataset(dsid, {'standard_name': 'solar_zenith_angle'})
+
+        for ds in [ds1, ds2, ds3, ds4]:
+            yield ds
+
+    def test_get_dataset(self):
+        """Retrieve datasets from a DNB file."""
+        for ds in self._dataset_iterator():
+            self.assertEqual(ds.shape, (752, 4064))
+            self.assertEqual(ds.dtype, np.float32)
+            self.assertEqual(ds.compute().shape, (752, 4064))
+            self.assertEqual(ds.attrs['rows_per_scan'], 16)
+
+    def test_distributed(self):
+        """Check that distributed computations work."""
+        from dask.distributed import Client
+        self.client = Client()
```

Comment:
Would it be better to have a separate class that always creates the client in setup rather than assigning to `self.client` here?

Reply:
When I end up having more tests, I will split the TestCase, yes. For now it's fine like it is I think